### ISSUE TITLE
将全连接层改成卷积层后模型不收敛了

### ISSUE THREAD

**Issue by hhk123** [YCG09/chinese_ocr#227]
想加速下的，就把最后的全连接层改成卷积层了．
参数，output的shape都是跟全连接一样的．
为啥loss一直不降．．．８００多很难受

def dense_cnn(input_x, n_class):
    _dropout_rate = 0.2
    _weight_decay = 1e-4
    _growth_rate = 8

    # conv 64 5*5 s=2
    x = Conv2D(64, (5, 5), strides=(2, 2), kernel_initializer='he_normal', padding='same',
               use_bias=False, kernel_regularizer=l2(_weight_decay))(input_x)

    # 8 layers dense block
    x = dense_block(x, 8, _growth_rate)
    # 128 outputs transition block
    x = transition_block(x, 128, dropout_rate=_dropout_rate, pool_type=1)

    # 8 layers dense block
    x = dense_block(x, 8, _growth_rate)
    # 128 outputs transition block
    x = transition_block(x, 128, dropout_rate=_dropout_rate, pool_type=1)

    # 8 layers dense block
    x = dense_block(x, 8, _growth_rate)
    x = BatchNormalization(axis=-1, epsilon=1.1e-5)(x)
    x = Activation('relu')(x)

    x20 = Permute((2, 1, 3), name='permute')(x)
    x30 = TimeDistributed(Flatten(), name='flatten')(x20)
    x40 = Conv1D(n_class, kernel_size=(1), strides=(1), padding='valid', 
                 bias_initializer=RandomNormal(mean=0.0, stddev=0.05, seed=None))(x30)
    x50 = Activation('softmax')(x40)
    return x50

**Comment by hhk123** [comment]
solved
change 
x40 = Conv1D(n_class, kernel_size=(1), strides=(1), padding='valid', 
             bias_initializer=RandomNormal(mean=0.0, stddev=0.05, seed=None))(x30)
x50 = Activation('softmax')(x40)
return x50
to 
x40 = Conv1D(n_class, kernel_size=(1), strides=(1), padding='valid', 
             bias_initializer=RandomNormal(mean=0.0, stddev=0.05, seed=None), activation='softmax')(x30)

return x40

**Comment by hhk123** [comment]
原来是前面的层数被锁了．．．无视吧．．．
